coral/src/app/features/topics/details/schema/components/SchemaPromotionModal.tsx

```diff
@@ -0,0 +1,81 @@
+import { Box, Switch, Textarea } from "@aivenio/aquarium";
+import { useState } from "react";
+import { Modal } from "src/app/components/Modal";
+
+interface SchemaPromotionModalProps {
+  onClose: () => void;
+  onSubmit: ({
+    remarks,
+    forceRegister,
+  }: {
+    remarks: string;
+    forceRegister: boolean;
+  }) => void;
+  isLoading: boolean;
+  targetEnvironment?: string;
+  version: number;
+  showForceRegister: boolean;
+}
+
+const SchemaPromotionModal = ({
+  onClose,
+  onSubmit,
+  isLoading,
+  targetEnvironment,
+  version,
+  showForceRegister,
+}: SchemaPromotionModalProps) => {
+  const [remarks, setRemarks] = useState<string>("");
+  const [forceRegister, setForceRegister] = useState(false);
+
+  return (
+    <Modal
+      title={`Promote schema to ${targetEnvironment}`}
+      close={onClose}
+      primaryAction={{
+        text: "Request schema promotion",
+        onClick: () =>
+          onSubmit({
+            remarks,
+            forceRegister,
+          }),
+        loading: isLoading,
+      }}
+      secondaryAction={{
+        text: "Cancel",
+        onClick: onClose,
+        disabled: isLoading,
+      }}
+    >
+      <Box display={"flex"} flexDirection={"column"} rowGap={"l1"}>
+        <p>
+          {`Promote the Version ${version} of the schema to ${targetEnvironment}?`}
+        </p>
+        {showForceRegister && (
+          <Switch
```

Comment:
I would think that a classic checkbox is more appropriated here. For me, a toogle implies an instant action - turn it on, something happens, turn it off, something happens. Like an electric switch basically, where there is an on and off state and I get an instant response which is which.

Also it always tends to be bit ambiguous to me (like: when is it "on" and "off"). When it's an instant action and therefor reversible, this not a big thing, because I can learn when it's "on" and "off" without risk.

But confirming something in this way, where it's not reversible and I have to submit this confirmation with other information, I think a checkbox is the better choice for usability and clearness. As far as I can know, that's also the general consensus with the "toggle vs switch" question.

Reply:
Sure, I was following the designs docs but I have no issue with either ^^

Reply:
🙏